File: ITServiceDesk.Client/src/types/device.ts
export enum DeviceStatus {
    Active = 1,
    Faulty = 2,
    Maintenance = 3,
    Storage = 4
}

export interface DeviceCategoryDto {
    id: string;
    name: string;
    description: string | null;
}

export interface DeviceDto {
    id: string;
    code: string;
    name: string;
    status: DeviceStatus;
    categoryId: string;
    categoryName: string;
    departmentId: string | null;
    departmentName: string | null;
    assignedUserId: string | null;
    assignedUserName: string | null;
}

export interface DeviceCreateDto {
    code: string;
    name: string;
    status: DeviceStatus;
    categoryId: string;
    departmentId: string | null;
    assignedUserId: string | null;
}

export interface DeviceUpdateDto {
    id: string;
    code: string;
    name: string;
    status: DeviceStatus;
    categoryId: string;
    departmentId: string | null;
    assignedUserId: string | null;
}

File: ITServiceDesk.Core/Entities/Device.cs
using ITServiceDesk.Core.Enums;

namespace ITServiceDesk.Core.Entities;

public class Device : BaseEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public DeviceStatus Status { get; set; } = DeviceStatus.Active;
    public DateTime? WarrantyExpirationDate { get; set; }
    
    
    // Navigation Properties
    public Guid CategoryId { get; set; }
    public DeviceCategory? Category { get; set; }
    
    public Guid? DepartmentId { get; set; }
    public Department? Department { get; set; }
    
    public Guid? AssignedUserId { get; set; }
    public AppUser? AssignedUser { get; set; }
    
    public ICollection<Ticket> Tickets { get; set; } = [];
}
